### ISSUE TITLE
Add CI for running .NET tests

### ISSUE THREAD

**Issue by sungam3r** [open-telemetry/opentelemetry-demo#748]
Surprised that CI checks were not running the tests 👎 .
@sungam3r -To unblock you progress here, please comment out the tests temporarily, create an issue to fix them (and assign to me). I hope this is okay with the maintainers of the repo to unblock temporarily.

_Originally posted by @cijothomas in https://github.com/open-telemetry/opentelemetry-demo/pull/746#discussion_r1107931240_

**Comment by puckpuck** [comment]
This issue is to help finish a now merged PR, and references a Closed PR. Closing this issue as such.

If we lack .NET test coverage, please open a new issue with specifics on what coverage we lack.
